source/Plugins/ExpressionParser/Swift/SwiftUserExpression.cpp

```diff
@@ -448,17 +443,20 @@ bool SwiftUserExpression::Parse(DiagnosticManager &diagnostic_manager,
     exe_scope = exe_ctx.GetTargetPtr();
   } while (0);
 
-  std::unique_ptr<ExpressionParser> parser(
-      new SwiftExpressionParser(exe_scope, *this, m_options));
+  m_parser =
+      llvm::make_unique<SwiftExpressionParser>(exe_scope, *this, m_options);
 
-  unsigned num_errors = parser->Parse(
-      diagnostic_manager, first_body_line, 
+  unsigned error_code = m_parser->Parse(
+      diagnostic_manager, first_body_line,
       first_body_line + source_code->GetNumBodyLines(), line_offset);
 
-  if (num_errors) {
+  if (error_code == 2) {
```

Comment:
I don't understand why these changes are in this patch, and I can't find a corresponding change to SwiftExpressionParser::Parse that would motivate them.

Reply:
This is the mechanism used to signal that we should tear down the scratch AST context build a per-module one and retry the expression, using the retry mechanism that exists for fix-its.